### PR TITLE
Support changing manual configuration while running

### PR DESCRIPTION
**Pull request by vishwamartur**
Related to #380

Add support for dynamic configuration updates while the application is running using a hot-reload approach.

* **ManualDiscovery (`exo/networking/manual/manual_discovery.py`)**
  - Add `start_file_watcher` method to initialize a file watcher using the `watchdog` library.
  - Add `reload_config` method to reload the network configuration dynamically.
  - Add `ConfigFileEventHandler` class to handle file modification events and trigger configuration reload.
  - Update `start` method to initialize the file watcher.
  - Update `stop` method to stop the file watcher.

* **NetworkTopology (`exo/networking/manual/network_topology_config.py`)**
  - Add `reload` method to reload the configuration from the file and update the `peers` attribute.

* **Main Application (`exo/main.py`)**
  - Import the `watchdog` library.

### CODE REVIEW

File: exo/networking/manual/network_topology_config.py [hunk 1]
```diff
@@ -29,3 +29,8 @@ def from_path(cls, path: str) -> "NetworkTopology":
       return cls.model_validate_json(config_data)
     except ValidationError as e:
       raise ValueError(f"Error validating network topology config from {path}: {e}") from e
+
+  def reload(self, path: str) -> None:
```
Comment (dtnewman) [inline exo/networking/manual/network_topology_config.py:33]: does this get called?

File: exo/networking/manual/manual_discovery.py [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import asyncio
 from exo.networking.discovery import Discovery
 from typing import Dict, List, Callable
+from watchdog.observers import Observer
+from watchdog.events import FileSystemEventHandler
```
Comment (dtnewman) [inline exo/networking/manual/manual_discovery.py:5]: needs to be added to dependencies

**Comment by AlexCheema** [comment]
Not a fan of adding a dependency on watchdog for this.
We don't need instant config application. Periodically checking every 5 secs for config changes is fine.

**Comment by AlexCheema** [comment]
Closing in favour of https://github.com/exo-explore/exo/pull/383
Please contact me for a $100 bounty with your ethereum address as I'd like to reward good experimentation: [alex@exolabs.net](mailto:alex@exolabs.net)

**Pull request closed without merging**